src/Features/Core/Portable/ExtractClass/ExtractClassWithDialogCodeAction.cs

```diff
@@ -0,0 +1,260 @@
+                    _selectedType.GetAttributes(),
+                    _selectedType.DeclaredAccessibility,
+                    _selectedType.GetSymbolModifiers(),
+                    TypeKind.Class,
+                    extractClassOptions.TypeName);
+
+                var containingNamespaceDisplay = namespaceService.GetContainingNamespaceDisplay(
+                    _selectedType,
+                    _document.Project.CompilationOptions);
+
+                // Add the new type to the solution. It can go in a new file or
+                // be added to an existing. The returned document is always the document
+                // containing the new type
+                var updatedDocument = extractClassOptions.SameFile
+                    ? await ExtractTypeHelpers.AddTypeToExistingFileAsync(
+                        symbolMapping.AnnotatedSolution.GetRequiredDocument(_document.Id),
+                        newType,
+                        symbolMapping,
+                        cancellationToken).ConfigureAwait(false)
+                    : await ExtractTypeHelpers.AddTypeToNewFileAsync(
+                        symbolMapping.AnnotatedSolution,
+                        containingNamespaceDisplay,
+                        extractClassOptions.FileName,
+                        _document.Project.Id,
+                        _document.Folders,
+                        newType,
+                        fileBanner,
+                        cancellationToken).ConfigureAwait(false);
+
+                // Update the original type to have the new base
+                var solutionWithUpdatedOriginalType = await GetSolutionWithBaseAddedAsync(
+                    updatedDocument.Project.Solution,
+                    symbolMapping,
+                    newType,
+                    extractClassOptions.MemberAnalysisResults,
+                    cancellationToken).ConfigureAwait(false);
+
+                // After all the changes, make sure we're using the most up to date symbol 
+                // as the destination for pulling members into
+                var documentWithTypeDeclaration = solutionWithUpdatedOriginalType.GetRequiredDocument(updatedDocument.Id);
+                newType = await GetNewTypeSymbolAsync(documentWithTypeDeclaration, newType, cancellationToken).ConfigureAwait(false);
+
+                // Use Members Puller to move the members to the new symbol
+                var finalSolution = await PullMembersUpAsync(
+                    solutionWithUpdatedOriginalType,
+                    newType,
+                    symbolMapping,
+                    extractClassOptions.MemberAnalysisResults,
+                    cancellationToken).ConfigureAwait(false);
+
+                return new[] { new ApplyChangesOperation(finalSolution) };
+            }
+            else
+            {
+                // If user click cancel button, options will be null and hit this branch
```

Comment:
that's seems super bizarre... fi the user hits cancel why are we doing anything beyond that point.

Reply:
This is a dialog cancel as opposed to some popup for work. Is that the confusion? Cancel from dialog still has to return something back from the options service. In this case I just used null